### ISSUE TITLE
Disable mount/remount of screen when navigating to the same screen in a StackNavigator

### ISSUE THREAD

**Issue by xanderberkein** [react-community/react-navigation#1953]
I want to wrap a container around some screens in my app that share some basic UI elements. As an example, I created two navigator routes that redirect to the same screen (the container with some shared ui), but with a different childComponent. The child component is the "real" screen, where the content of the screen will be.

<img src="https://user-images.githubusercontent.com/10937632/27383291-a2af802c-568a-11e7-8618-b0163eac2b7a.png" width="250">

The problem is that the HomeContainer gets unmounted and remounted when changing screen, destroying its state. Is there anyway to wrap UI over components, without destroying the state of the container when switching screens?

This is my Navigator:

```javascript
const MainNavigator = StackNavigator(
  {
    Home: {
      screen: props => <HomeContainer {...props} screenProps={{component: ChildComponent}} />,
      navigationOptions: { title: 'HomeScreen' }
    },
    AlternativeHome: {
      screen: props => <HomeContainer {...props} screenProps={{component: ChildComponent2}} />,
      navigationOptions: { title: 'Other Screen' }
    },
 });
```

These are the ChildComponents, the screens where the main content will be:

```javascript
const ChildComponent = (
  <View>
    <Text>I am a childcomponent</Text>
    <Text>This is the "real screen"</Text>
  </View>
);
const ChildComponent2 = (
  <View>
    <Text>I am a alternative childcomponent</Text>
    <Text>Haha</Text>
  </View>
);
```

The HomeContainer renders the shared UI and returns the component:
```javascript
render() {
  return (
    <View>
      <Text>I am the Container</Text>
      <Text>This is some shared UI between all screens</Text>
      <Text>Below is my childComponent</Text>
      <Card>
        {this.props.screenProps.component}
      </Card>
      <Button title="Go to Other Screen" onPress={() => this.props.navigation.navigate('AlternativeHome')} />
    </View>
  );
}
```

The screenshot above is the result of this code.

So, to recap, when I press the button to navigate to the other screen, the HomeContainer unmounts and remounts again. I would like it to stay mounted.

If there is another way to have shared, persistent UI accross screens, I'm open to suggestions. I tried multiple things already, but the above solution is the closest I came to what I want.

**Comment by xanderberkein** [comment]
To give some insight in another approach to solve this issue, here is something else I tried.

I wrapped the screens that share the same ui elements in a `HomeContainer`

```javascript
const MainNavigator = StackNavigator(
  {
    Home: {
      screen: HomeContainer,
      navigationOptions: { title: 'Home' },
    },
    Login: {
      screen: LoginScreen,
      navigationOptions: { header: false },
    }
);
```

Then, inside the HomeContainer, I put the shared UI components, along with a new Navigator:

```javascript
navigateTo(routeName) {
  const nav = NavigationActions.navigate({
    routeName: 'Home',
    action: NavigationActions.navigate({ routeName: 'Home2' }),
  });
  this.props.navigation.dispatch(nav);
}

render() {
  const HomeNavigator = StackNavigator(
    {
      Home: {
        screen: MessageScreen,
        navigationOptions: { header: false }
      },
      Home2: {
        screen: MessageScreen2,
        navigationOptions: { header: false },
      },
    }
  );

  return (
    <View style={{ flex: 1 }}>
      <Text>I am the Container</Text>
      <Text>This is some shared UI between all screens</Text>
      <Text>Below is my childComponent</Text>
      <HomeNavigator />
      <Button title="Go to Other Screen" onPress={this.navigateTo.bind(this, 'Home2')} />
    </View>
  );
}

The problem here is that I can't get the button to work. I used the solution of #1401[(direct link to comment)](https://github.com/react-community/react-navigation/issues/1401#issuecomment-299699164), but this doesn't seem to work since HomeNavigator is no direct child of MainNavigator: there is a Container in between.

Any ideas on this one?

**Comment by spencercarli** [comment]
Hi! In an effort to get the hundreds of issues on this repo under control I'm closing issues tagged as questions. Please don't take this personally - it's simply something we need to do to get the issues to a manageable point. If you still have a question I encourage you to re-read the docs, ask on StackOverflow, or ask on the #react-navigation Reactiflux channel. Thank you!

**Comment by amirmatin** [comment]
@spencercarli  why these issues and similar ones keep getting closed without any clear solutions or updates ? 
@xanderberkein  did you manage to find a solution for this ? I am facing a similar issue and very annoyed as it seems all the similar threads are being closed with no solution ...

**Comment by xanderberkein** [comment]
No, I didn't find a solution, I just redesigned the app to use a less complicated routes layout. Far from ideal, but the other navigation I tried (wix navigation) didn't meet my needs either.

**Comment by amirmatin** [comment]
@xanderberkein  Thanks for your feedback.. I am giving the "react-native-navigation" a shot which I think is also called wix ! ...  will let you know here if I found a solution ...

**Comment by shekharskamble** [comment]
@amirmatin did you find any solution, I am having the same issue and parent screen gets unmount/mount again, clearing all of its state.....

**Comment by amirmatin** [comment]
@shekharskamble  I switched to react-native-navigation, it does not have that issue, however it is abit fiddly to get it setup on iOS. they also have an example page which is very handy...

this is the link:  https://github.com/wix/react-native-navigation
hope this helps !

**Comment by shekharskamble** [comment]
Thanks @amirmatin
